tests/testthat/test-DataValidation.R

```diff
@@ -0,0 +1,108 @@
+  data <- CleanData(read.csv("Cont_long.csv"))
+  
+  validation_result <- ValidateUploadedData(data, "Continuous")
+  
+  expect_true(validation_result$valid)
+  expect_equal(validation_result$message, "Data is valid")
+})
+
+test_that("ValidateUploadedData() identifies valid continuous wide data", {
+  data <- CleanData(read.csv("Cont_wide.csv"))
+  
+  validation_result <- ValidateUploadedData(data, "Continuous")
+  
+  expect_true(validation_result$valid)
+  expect_equal(validation_result$message, "Data is valid")
+})
+
+test_that("ValidateUploadedData() identifies valid binary long data", {
+  data <- CleanData(read.csv("Binary_long.csv"))
+  
+  validation_result <- ValidateUploadedData(data, "Binary")
+  
+  expect_true(validation_result$valid)
+  expect_equal(validation_result$message, "Data is valid")
+})
+
+test_that("ValidateUploadedData() identifies valid binary wide data", {
+  data <- CleanData(read.csv("Binary_wide.csv"))
+  
+  validation_result <- ValidateUploadedData(data, "Binary")
+  
+  expect_true(validation_result$valid)
+  expect_equal(validation_result$message, "Data is valid")
+})
+
+test_that("ValidateUploadedData() identifies invalid coninuous long data, missing colunm 'N'", {
+  data <- CleanData(read.csv("invalid_data/continuous-missing-long.csv"))
+  
+  validation_result <- ValidateUploadedData(data, "Continuous")
+  
+  expect_false(validation_result$valid)
+  expect_equal(validation_result$message, "Missing columns for Continuous data: N")
+})
+
+test_that("ValidateUploadedData() identifies invalid coninuous wide data, missing all colunms 'N.*'", {
+  data <- CleanData(read.csv("invalid_data/continuous-missing-entirely-wide.csv"))
+  
+  validation_result <- ValidateUploadedData(data, "Continuous")
+  
+  expect_false(validation_result$valid)
+  expect_equal(validation_result$message, "Missing columns for Continuous data: N")
+})
+
+test_that("ValidateUploadedData() identifies invalid coninuous wide data, missing some colunms 'N.*'", {
+  data <- CleanData(read.csv("invalid_data/continuous-missing-partially-wide.csv"))
+  
+  validation_result <- ValidateUploadedData(data, "Continuous")
+  
+  expect_false(validation_result$valid)
+  expect_equal(validation_result$message, "For wide format data, numbered columns (T, N, Mean, SD) must all have matching sequential indices, starting from 1")
+})
+
+test_that("ValidateUploadedData() identifies invalid coninuous wide data, with inconsistently numbered colunms 'N.*'", {
+  data <- CleanData(read.csv("invalid_data/continuous-misnumbered-wide.csv"))
+  
+  validation_result <- ValidateUploadedData(data, "Continuous")
+  
+  expect_false(validation_result$valid)
+  expect_equal(validation_result$message, "For wide format data, numbered columns (T, N, Mean, SD) must all have matching sequential indices, starting from 1")
+})
+
+test_that("ValidateUploadedData() identifies invalid coninuous long data, missing colunm 'N'", {
```

Comment:
Should be binary, also in several other places.

Reply:
Fixed